apis/datadoghq/v2alpha1/datadogagent_types.go

```diff
@@ -38,9 +38,155 @@ type DatadogAgentSpec struct {
 }
 
 // DatadogFeatures are Features running on the Agent and Cluster Agent.
-// TODO add the features (APM, Kubernetes State Metrics Core...)
 // +k8s:openapi-gen=true
-type DatadogFeatures struct{}
+type DatadogFeatures struct {
+	// Application-level features
+	// APM (Application Performance Monitoring) configuration
+	APM *APMFeatureConfig `json:"apm,omitempty"`
+	// Logs configuration
+	Logs *LogsFeatureConfig `json:"logs,omitempty"`
+	// Process configuration
+	Process *ProcessFeatureConfig `json:"process,omitempty"`
+	// CWS (Cloud Workload Security) configuration
+	CWS *CWSFeatureConfig `json:"cws,omitempty"`
+	// CWS (Cloud Security Posture Management) configuration
+	CSPM *CSPMFeatureConfig `json:"cspm,omitempty"`
+	// NPM (NetworkPerformanceMonitoring) configuration
+	NPM *NPMFeatureConfig `json:"npm,omitempty"`
+
```

Comment:
Another feature do discuss with the Network monitoring team is the new [UnifiedServiceMonitoring](https://www.datadoghq.com/blog/dash-2021-new-feature-roundup/#universal-service-monitoring) 

What I understood it is separated from the NPM. so maybe we will need to add an "USM"